File: codingame/puzzles/dontPanic/dontPanic1.js
var inputs = readline().split(' ');
var nbFloors = parseInt(inputs[0]);
var width = parseInt(inputs[1]);
var nbRounds = parseInt(inputs[2]);
var exitFloor = parseInt(inputs[3]); 
var exitPos = parseInt(inputs[4]);
var nbTotalClones = parseInt(inputs[5]);
var nbAdditionalElevators = parseInt(inputs[6]);
var nbElevators = parseInt(inputs[7]);

printErr('number of elevators...', nbElevators);

var elevatorPos = [];
for (var i = 0; i < nbElevators; i++) {
    var inputs = readline().split(' ');
    var elevatorFloor = parseInt(inputs[0]);
    elevatorPos[elevatorFloor] = parseInt(inputs[1]);
}
var currentFloor = 0;
while (true) {
    var inputs = readline().split(' ');
    var cloneFloor = parseInt(inputs[0]);
    var clonePos = parseInt(inputs[1]);
    var direction = inputs[2];
    var cmd = "WAIT";
    if(cloneFloor == currentFloor) {
        var targetPos = (exitFloor == cloneFloor)? exitPos: elevatorPos[cloneFloor];
        if((direction == "RIGHT" && clonePos > targetPos) ||
            (direction == "LEFT" && clonePos < targetPos)) {
            cmd = "BLOCK";
        }
        currentFloor++;
    }
    print(cmd);
}

File: codingame/puzzles/dontPanic/dontPanic1.rs
use std::io;

macro_rules! parse_input {
    ($x:expr, $t:ident) => ($x.trim().parse::<$t>().unwrap())
}

fn r() -> Vec<String> {
    let mut input_line = String::new();
    io::stdin().read_line(&mut input_line).unwrap();
    input_line.split(" ").map(|s| s.to_string()).collect()
}

fn main() {
    let inputs = r();
    let exit_floor = parse_input!(inputs[3], i32);
    let exit_pos = parse_input!(inputs[4], i32);
    let nb_elevators = parse_input!(inputs[7], i32);
    let mut elevator_p = [0;100];
    for i in 0..nb_elevators as usize {
        let inputs = r();
        elevator_p[parse_input!(inputs[0], usize)] = parse_input!(inputs[1], i32);
    }

    let mut current_floor = 0;
    loop {
        let inputs = r();
        let clone_floor = parse_input!(inputs[0], i32);
        let clone_pos = parse_input!(inputs[1], i32);
        let direction = inputs[2].trim().to_string();
        let mut cmd = "WAIT";
        if clone_floor == current_floor {
            let target_pos = if exit_floor == clone_floor { exit_pos } else { elevator_p[clone_floor as usize] };
            if direction == "RIGHT" && clone_pos > target_pos || direction == "LEFT" && clone_pos < target_pos {
                cmd = "BLOCK";
            }
            current_floor += 1;
        }
        println!("{}", cmd);
    }
}
